File: dotnet/src/Microsoft.Agents.Validation/Validators/ContentValidators/EventContentValidator.cs
using Microsoft.Agents.Xml.Generated.Models;

namespace Microsoft.Agents.Xml.Validation.Validators.ContentValidators;

/// <summary>
/// Validator for EventContent.
/// </summary>
public class EventContentValidator : ContentValidatorBase<EventContent>
{
    public override ValidationResult Validate(EventContent content, ValidationContext? context = null)
    {
        var errors = new List<ValidationError>();

        // CNT-015: EventContent name must be non-empty
        var nameError = ValidateNotEmpty(content.Name, "Name",
            ValidationErrorCode.CNT_015,
            "EventContent name must be non-empty");
        if (nameError != null)
            errors.Add(nameError);

        return new ValidationResult(errors);
    }
}

File: typescript/packages/agents-protocol-validation/src/validators/content/EventContentValidator.ts
import type { EventContent } from '@microsoft/agents-protocol-abstractions';
import type { ValidationContext } from '../../core/ValidationContext';
import { ContentValidatorBase } from '../base/ContentValidatorBase';
import { ValidationResult } from '../../core/ValidationResult';
import { ValidationErrorCode } from '../../core/ValidationErrorCode';

/**
 * Validator for EventContent.
 */
export class EventContentValidator extends ContentValidatorBase<EventContent> {
  public validate(content: EventContent, context?: ValidationContext): ValidationResult {
    const errors = this.collectErrors(
      // CNT-015: event name must be non-empty
      this.validateNotEmpty(
        content.name,
        'name',
        ValidationErrorCode.CNT_015,
        'EventContent name must be non-empty'
      )
    );

    return new ValidationResult(errors);
  }
}
